test/distributed/org/apache/cassandra/distributed/mock/nodetool/InternalNodeProbe.java

```diff
@@ -0,0 +1,202 @@
+    {
+        return new EndpointSnitchInfo();
+    }
+
+    @Override
+    public CacheServiceMBean getCacheServiceMBean()
+    {
+        return cacheService;
+    }
+
+    @Override
+    public void repairAsync(final PrintStream out, final String keyspace, Map<String, String> options) throws IOException
+    {
+        RepairRunner runner = new RepairRunner(out, ssProxy, keyspace, options);
+        try
+        {
+            ssProxy.addNotificationListener(runner, null, null);
+            runner.run();
+        }
+        catch (Exception e)
+        {
+            throw new IOException(e) ;
+        }
+        finally
+        {
+            try
+            {
+                ssProxy.removeNotificationListener(runner);
+            }
+            catch (Throwable e)
+            {
+                out.println("Exception occurred during clean-up. " + e);
+            }
+        }
+    }
+
+    @Override
+    // simply resume bootstrap, instead of streaming the JMX notification to stdout
+    public void resumeBootstrap(PrintStream out) throws IOException
+    {
+        if (ssProxy.resumeBootstrap())
+        {
+            out.println("Resuming bootstrap");
+        }
+        else
+        {
+            out.println("Node is already bootstrapped.");
+        }
+    }
+
+    // dtest runs with disable_mbean_registration. No ColumnFamilyStoreMBean is registered, thus not supporting get the MBean.
+    @Override
+    public ColumnFamilyStoreMBean getCfsProxy(String ks, String cf)
+    {
+        return Keyspace.open(ks).getColumnFamilyStore(cf);
+    }
+
+    @Override
+    // The below methods are only used by the commands (i.e. Info, TableHistogram, TableStats, etc.) that display informations. Not useful for dtest, so disable it.
+    public Object getCacheMetric(String cacheType, String metricName)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Iterator<Map.Entry<String, ColumnFamilyStoreMBean>> getColumnFamilyStoreMBeanProxies()
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Multimap<String, String> getThreadPools()
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Object getThreadPoolMetric(String pathName, String poolName, String metricName)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Object getColumnFamilyMetric(String ks, String cf, String metricName)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CassandraMetricsRegistry.JmxTimerMBean getProxyMetric(String scope)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Object getCompactionMetric(String metricName)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public long getStorageMetric(String metricName)
+    {
+        throw new UnsupportedOperationException();
```

Comment:
FYI `org.apache.cassandra.metrics.StorageMetrics`. Though the special casing would be annoying.